### PR TITLE
Attribute Picker: Only allow selectable attributes in Drop down

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Only allow certain attributes in the drop down, while still allowing for custom values

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
I can't seem to get Vaadin to work with the formatted value and allow custom values... Any ideas?

Right now it works if you type the attr in the code. Which maybe we want. And then we could just disable the picker if its not in the select box.

But my goal would be to allow any text in the field that would be the attribute config value but also allow them to choose from a dropdown of the selectable values

File: src/components/entity/ha-entity-attribute-picker.ts [hunk 1]
```diff
@@ -137,6 +172,19 @@ class HaEntityAttributePicker extends LitElement {
...
   }
 
+  private _selectableAttributes = memoizeOne((entity: string) => {
+    const stateObj = this.hass.states[entity];
+    if (!stateObj) {
+      return [];
+    }
+
+    return Object.keys(stateObj.attributes).filter((attr) =>
+      SELECTABLE_ATTRIBUTES[entity.substring(0, entity.indexOf("."))].includes(
```
Comment (bramkragten) [inline src/components/entity/ha-entity-attribute-picker.ts:182]:
```suggestion
	const domain = computeDomain(entity);
    return Object.keys(stateObj.attributes).filter((attr) =>
      SELECTABLE_ATTRIBUTES[domain].includes(
```

**Comment by bramkragten** [comment]
> I can't seem to get Vaadin to work with the formatted value and allow custom values... Any ideas?
> 
> Right now it works if you type the attr in the code. Which maybe we want. And then we could just disable the picker if its not in the select box.
> 
> But my goal would be to allow any text in the field that would be the attribute config value but also allow them to choose from a dropdown of the selectable values

🤔 Why isn't it working what is it doing?

**Comment by zsarnett** [comment]
@bramkragten I just don't know if it is possible. When you allow custom values the would expect to be able to select from the list and just input what you want without selecting. But it doesn't seem to allow me to input a value that is custom. The changed even never fires unless I click an option

**Comment by bramkragten** [comment]
`allowCustomValue` should take care of that:
https://vaadin.com/components/vaadin-combo-box/html-api/elements/Vaadin.ComboBoxLightElement#property-allowCustomValue

**Comment by zsarnett** [comment]
Right. I have that... Doesnt work when putting the value through a formatter

**Comment by bramkragten** [comment]
You mean it works when changing:
```ts
.value=${this._value ? formatAttributeName(this._value) : ""}
```
in
```ts
.value=${this._value"}
```

**Comment by zsarnett** [comment]
@bramkragten Okay so I found the issue with the cutom value it wasn't the formatting

But now there is a different issue

https://i.gyazo.com/1e8eb27c84f7f638c9a1ddbd933ce7e2.mp4

Long video

**Comment by zsarnett** [comment]
The picker doesnt like the friendly version of the name in the input. And the friendly version doesn't persist if you click in and out of the box

**Comment by zsarnett** [comment]
I wonder if I can just send in an object instead of just a string

**Pull request closed without merging**